splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java

```diff
@@ -428,10 +459,24 @@ public void reopenCore() throws StandardException{
     public ExecRow getNextRowCore() throws StandardException{
         try{
             if(execRowIterator.hasNext()){
-                locatedRow= execRowIterator.next();
-                if(LOG.isTraceEnabled())
-                    SpliceLogUtils.trace(LOG,"getNextRowCore %s locatedRow=%s",this,locatedRow);
-                return locatedRow;
+                try {
+                    locatedRow = execRowIterator.next();
+                    if (LOG.isTraceEnabled())
+                        SpliceLogUtils.trace(LOG, "getNextRowCore %s locatedRow=%s", this, locatedRow);
+                    ExecRow result = locatedRow;
+                    returnedRows = true;
+                    return result;
+                } catch (ResubmitDistributedException e) {
+                    if (!returnedRows) {
+                        LOG.warn("The query consumed too many resources running in control mode, resubmitting in Spark");
```

Comment:
If I understand your question correctly, no, it's not possible: as soon as the control query returns some values we can no longer resubmit to Spark, we raise an exception and ask the user to submit to Spark himself. That's guarded by the `returnedRows` field.